dali/pipeline/util/thread_pool.cc

```diff
@@ -38,7 +38,12 @@ ThreadPool::ThreadPool(int num_thread, int device_id, bool set_affinity)
 
 ThreadPool::~ThreadPool() {
   // Wait for work to find errors
-  WaitForWork(false);
+  // terminate explicitly as we are finishing the program anyway
+  try {
```

Comment:
This replaces one line of code with five that do exactly the same thing...

Reply:
But we do it explicitly and static analyzer was complaining about this.